### PR TITLE
feat (new llm): add support for openrouter

### PR DESCRIPTION
**Pull request by Korayem**
Code mainly copied from togetherai llm provider

Covers feature request #2679

### CODE REVIEW

**Comment by Yeuoly** [comment]
It would be better to consider changing the icon, as the current ones do not align well with our overall style, and the square icon seems to be completely transparent.
image
image

**Comment by Korayem** [comment]
> @Yeuoly  It would be better to consider changing the icon, as the current ones do not align well with our overall style, and the square icon seems to be completely transparent. 

Check my last commit, I revised both
image

File: api/core/model_runtime/model_providers/openrouter/openrouter.yaml [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  en_US: openrouter_square.svg
+icon_large:
+  en_US: openrouter.svg
+background: "#F1EFED"
+help:
+  title:
+    en_US: Get your API key from openrouter.ai
+    zh_Hans: 从 openrouter.ai 获取 API Key
+  url:
+    en_US: https://api.openrouter.ai/
```
Comment (Yeuoly) [inline api/core/model_runtime/model_providers/openrouter/openrouter.yaml]: This URL does not work for me, is it https://openrouter.ai/keys?

**Review by Yeuoly — CHANGES_REQUESTED** [review]

File: api/core/model_runtime/model_providers/openrouter/openrouter.yaml (hunk 1 above)
Comment (Korayem) [reply api/core/model_runtime/model_providers/openrouter/openrouter.yaml]: oh you're right! sorry for that... fixed it in the new commit

Comment (Yeuoly) [reply api/core/model_runtime/model_providers/openrouter/openrouter.yaml]: Emmm, it looks like it's incorrect also
image

Comment (Korayem) [reply api/core/model_runtime/model_providers/openrouter/openrouter.yaml]: Sorry I added `/key` but didn't remove the subdomain part of the url

Please check now

**Review by Yeuoly — APPROVED** [review]
LGTM

**Pull request merged**